static/app/utils/profiling/profile/chromeTraceProfile.tsx

```diff
@@ -0,0 +1,290 @@
+/**
+ * The import code is very similar to speedscope's import code. The queue approach works well and allows us
+ * to easily split the X events and handle them. There are some small differences when it comes to building
+ * profiles where we opted to throw instead of closing a frame that was never opened.
+ *
+ * Overall, it seems that mostly typescript compiler uses this output, so we could possibly do a bit more
+ * in order to detect if this is a tsc trace and mark the different compiler phases and give users the preference
+ * to color encode by the program/bind/check/emit phases.
+ */
+
+import {Frame} from 'sentry/utils/profiling/frame';
+import {Profile} from 'sentry/utils/profiling/profile/profile';
+
+import {EventedProfile} from './eventedProfile';
+import {ProfileGroup} from './importProfile';
+
+export function isChromeTraceFormat(input: any): input is ChromeTrace.ProfileType {
+  return isChromeTraceArrayFormat(input) || isChromeTraceObjectFormat(input);
+}
+
+function isChromeTraceObjectFormat(input: any): input is ChromeTrace.ObjectFormat {
+  return typeof input === 'object' && 'traceEvents' in input;
+}
+
+function isChromeTraceArrayFormat(input: any): input is ChromeTrace.ArrayFormat {
+  return Array.isArray(input);
+}
+
+export function importChromeTrace(input: string | ChromeTrace.ProfileType): ProfileGroup {
+  if (isChromeTraceObjectFormat(input)) {
+    throw new Error('Chrometrace object format is not yet supported');
+  }
+  if (isChromeTraceArrayFormat(input)) {
+    return parseChromeTraceArrayFormat(input);
+  }
+
+  throw new Error('Failed to parse trace input format');
+}
+
+type ProcessId = number;
+type ThreadId = number;
+
+export function splitEventsByProcessAndTraceId(
+  trace: ChromeTrace.ArrayFormat
+): Record<ProcessId, Record<ThreadId, ChromeTrace.Event[]>> {
+  const collections: Record<ProcessId, Record<ThreadId, ChromeTrace.Event[]>> = {};
+
+  for (let i = 0; i < trace.length; i++) {
+    if (typeof trace[i].pid !== 'number') {
+      continue;
+    }
+    if (typeof trace[i].tid !== 'number') {
+      continue;
+    }
+
+    const event = trace[i];
```

Comment:
may as well initialize `event` at the beginning of the loop so you can use it in the `if` conditions